tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -857,6 +858,8 @@ void ROOT::Experimental::Internal::RPagePersistentSink::UpdateSchema(const RNTup
    }
 
    const auto nColumns = descriptor.GetNPhysicalColumns();
+   fOpenColumnRanges.reserve(fOpenColumnRanges.size() + (nColumns - nColumnsBeforeUpdate));
+   fOpenPageRanges.reserve(fOpenPageRanges.size() + (nColumns - nColumnsBeforeUpdate));
```

Comment:
Do these really provide any measurable improvement? IMHO it should at least be a separate commit, maybe even a separate PR

Reply:
Reserving an array when the size is known a priori is just the correct thing to do and there should be a very good reason *not* to do so, rather than having to justify doing it.  I don't agree with the fact that these hinder readability. Otherwise, for the same reasoning, we should provide measurements every time we choose to pass a `const std::string &` rather than a `std::string` and similar.

Reply:
We can agree to disagree. It's an extra line with an argument that you need to use brain power to understand and hopefully get right, where in many cases it doesn't buy you anything measurable. I think the comparison to passing by reference doesn't make sense because that one a) is the same line, so it's not extra, and b) there is a semantic difference between passing a const-ref and creating a copy.